### PR TITLE
fix(Auth): Fixing handling of service SdkError in Auth tasks

### PR DESCRIPTION
**Pull request by harsh62**
## Issue \#
#2857 

## Description
Some of the Auth Tasks were not catching the SdkErrors correctly. Added correct handling and tests. 

## General Checklist

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [x] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] New or updated tests include `Given When Then` inline code documentation and are named accordingly `testThing_condition_expectation()`
- [ ] If breaking change, documentation/changelog update with migration instructions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jcjimenez — APPROVED** [review]
LGTM

File: AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UserTasks/AWSAuthUpdateUserAttributesTask.swift [hunk 1]
```diff
@@ -36,6 +37,8 @@ class AWSAuthUpdateUserAttributesTask: AuthUpdateUserAttributesTask {
             return try await updateUserAttribute(with: accessToken)
         } catch let error as UpdateUserAttributesOutputError {
             throw error.authError
+        } catch let error as SdkError<UpdateUserAttributesOutputError> {
+            throw error.authError
         } catch let error as AuthError {
             throw error
```
Comment (royjit) [inline AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UserTasks/AWSAuthUpdateUserAttributesTask.swift:43]:
```suggestion
        } catch let error as AuthErrorConvertible {
            throw error.authError
        }
```

Can we remove the two catch and do this?

Comment (harsh62) [reply AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/UserTasks/AWSAuthUpdateUserAttributesTask.swift]: Thanks for pointing it out.. Didn't realize that `SdkError` was conforming to `AuthErrorConvertible`. Fixed in the update.

**Review by royjit — APPROVED** [review]
LGTM

**Pull request merged**